dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -658,15 +659,29 @@ public Token addOrcidToken(@ApiParam(hidden = true) @Parameter(hidden = true, na
             accessToken = tokenResponse.getAccessToken();
             refreshToken = tokenResponse.getRefreshToken();
 
-            // ORCID API returns the user's ORCID username in the "name" property
+            // ORCID API returns the username and orcid id along with the tokens
+            // get them to store in the token and user tables
             username = tokenResponse.get("name").toString();
+            orcid = tokenResponse.get("orcid").toString();
 
         } catch (IOException e) {
             LOG.error("Retrieving accessToken was unsuccessful" + e.getMessage());
             throw new CustomWebApplicationException(e.getMessage(), HttpStatus.SC_BAD_REQUEST);
         }
 
         if (user != null) {
+            // save the ORCID to the enduser table
```

Comment:
Included two possible solutions here. First is adding an ORCID column to the enduser table and storing it there. Second is storing it in a user profile.

Reply:
I like adding it to the user profile.